File: app/src/androidTest/java/com/dream/freshnews/SourcesActivityTest2.java
package com.dream.freshnews;

/**
 * Created by lixingming on 25/10/2017.
 */

import androidx.test.espresso.assertion.ViewAssertions;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import com.dream.freshnews.sources.SourcesActivity;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

@RunWith(AndroidJUnit4.class)
public class SourcesActivityTest2 {
    public SourcesActivityTest2() {
        super();
    }

    @Rule
    public ActivityTestRule<SourcesActivity> mActivityRule = new ActivityTestRule<>(SourcesActivity.class);

    @Before
    public void setup() {

    }

    @Test
    public void testTextViewDisplay() {
        onView(withId(R.id.menu_refresh)).check(ViewAssertions.matches(isDisplayed()));
    }

    @Test
    public void testRefresh() {
        onView(withId(R.id.menu_refresh)).perform(click());

        onView(withId(R.id.rv_sources)).perform(actionOnItemAtPosition(0, click()));

        onView(withId(R.id.rv_top_headlines)).check(ViewAssertions.matches(isDisplayed()));
    }

}

File: app/src/androidTest/kotlin/com/dream/freshnews/SourcesActivityTest2.kt
package com.dream.freshnews

import androidx.recyclerview.widget.RecyclerView
import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.contrib.RecyclerViewActions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import androidx.test.rule.ActivityTestRule
import com.dream.freshnews.sources.SourcesActivity
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Created by lixingming on 25/10/2017.
 */
@RunWith(AndroidJUnit4::class)
@LargeTest
class SourcesActivityTest2 {
    @get:Rule
    var mActivityRule = ActivityTestRule(SourcesActivity::class.java)

    @Before
    fun setup() {
    }

    @Test
    fun testTextViewDisplay() {
        Espresso.onView(ViewMatchers.withId(R.id.menu_refresh))
            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
    }

    @Test
    fun testRefresh() {
        Espresso.onView(ViewMatchers.withId(R.id.menu_refresh)).perform(ViewActions.click())
        Espresso.onView(ViewMatchers.withId(R.id.rv_sources)).perform(
            RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
                0,
                ViewActions.click()
            )
        )
        Espresso.onView(ViewMatchers.withId(R.id.rv_top_headlines))
            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
    }
}